### ISSUE TITLE
[TW-449] concurrent task add commands can silently fail

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#484]
_steve rader on 2012-11-30T12:48:25Z says:_ 

 I set up cron to add two tasks at the same time, 4am this morning.
The 2nd didn't get added.  ["Wow!", right?]
I'm NOT immediately able to reproduce the error.
I'll try harder later today else tomorrow.
I'm not sure about additional steps would useful for getting this fixed.
Assuming I can, please advise: what debugging etc would be helpful??

<pre>
$ show 41
(Message inbox:41)
Date:    Fri, 30 Nov 2012 04:00:01 CST
To:      rader(at)chive.hep.wisc.edu
From:    root(at)chive.hep.wisc.edu (Cron Daemon)
Subject: Cron <rader(at)chive> task add Concurrent test B

Created task 50.

$ show 42
(Message inbox:42)
Date:    Fri, 30 Nov 2012 04:00:01 CST
To:      rader(at)chive.hep.wisc.edu
From:    root(at)chive.hep.wisc.edu (Cron Daemon)
Subject: Cron <rader(at)chive> task add Concurrent test A

Created task 50.

$ task ls concurrent

ID Proj Prio Age Description
50            2h Concurrent test A

1 task
</pre>

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2012-11-30T12:48:25Z
Modified: 2014-02-09T01:54:31Z
```

**Comment by taskwarrior** [comment]
_steve rader on 2012-11-30T18:11:00Z says:_ 

 Well, I put in a every minute cron job testing this...
The rate of reproduction is about 0.6% (2 in 310 tries), 
and I wrote a stupid shell script to avoid it...
so... nevermind?!
<pre>
$ cat ~/bin/task-batch
1. !/bin/bash
if [ $# != 1 ]; then
  echo usage: task-batch FILE
  exit 1
fi
f=$1;
if [ ! -f "$f" ]; then
  f=$HOME/etc/$f
fi
if [ ! -f "$f" ]; then
  echo task-batch: $f: file not found
  exit 1
fi
IFS='
'
for l in `cat $f`; do
  if [ "$l" != "" ]; then
    task add $l > /dev/null 2>&1
  fi
done
<pre>

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2012-12-01T16:58:45Z says:_ 

 Steve, I am at a loss as to how to proceed on this one.  Not ignoring it - just stumped.

**Comment by taskwarrior** [comment]
_steve rader on 2012-12-01T17:21:59Z says:_ 

 Once I complained about a squeak with the interior of my car, and my irascible
mechanic, whom I love, said: "well, then turn up the f*cking radio!"

I think the way to proceed, if you really want to, is for you to put in a
bunch of debug msgs around the file locking and file I/O.  Then I could run
with rc.debug=on and perhaps send along some perhaps meaningful information.

But really, I think this is a situation where it really is appropriate
to just say "don't do that"--especially given that there's a perfectly good
remedy available.  No need for a cure.

[Sorry, when I filed this report I ASSUMED this bug was easily reproducible
and thus a really bad one.]
